srcts/src/main.ts

```diff
@@ -4155,8 +4155,28 @@ function main(): void {
     }
 
     if (dep.script && !restyle) {
-      const scripts = $.map(asArray(dep.script), function (scriptName) {
-        return $("<script>").attr("src", href + "/" + encodeURI(scriptName));
+      const scripts_attrs = asArray(dep.script);
+      const scripts = $.map(scripts_attrs, function (x) {
+        let script = document.createElement("script");
+
+        // htmlDependency()'s script arg may be a list of attributes
+        if (typeof x === "string") {
+          x = { src: x };
+        }
+        const attrs = Object.keys(x);
+
+        for (let i = 0; i < attrs.length; i++) {
+          const attr = attrs[i];
+          let val = x[attr];
+
+          if (attr === "src") {
+            val = href + "/" + encodeURI(val);
+          }
+          // If val isn't truthy (e.g., null), consider it a boolean attribute
+          script.setAttribute(attr, val ? val : "");
+        }
```

Comment:
I think you may be able to replace this with:

```suggestion
        for (let [attr, val] of Object.entries(x)) {
          if (attr === "src") {
            val = href + "/" + encodeURI(val);
          }
          // If val isn't truthy (e.g., null), consider it a boolean attribute
          script.setAttribute(attr, val ? val : "");
        }
```

See https://effectivetypescript.com/2020/05/26/iterate-objects/

@schloerke, @nstrayer, it would be great if you guys could weigh in on this.

Reply:
A polyfill for `Object.entries` will be required (IE / Phantom), but it is more succinct

(Changed the proposed to a `const` instead of a `let` in the for loop.)

```suggestion
        for (const [attr, val] of Object.entries(x)) {
          if (attr === "src") {
            val = href + "/" + encodeURI(val);
          }
          // If val isn't truthy (e.g., null), consider it a boolean attribute
          script.setAttribute(attr, val ? val : "");
        }
```

Reply:
With `const [attr, val]`, isn't that a problem for `val = href + "/" + encodeURI(val)`?

Reply:
Also, IMO, it's not worth adding a polyfill for this.